src/core/python/utility/pyboost_cv_converter.cpp

```diff
@@ -435,7 +285,60 @@ void matFromNDArrayBoostConverter::construct(
     data->convertible = storage;
 }
 
+// pt converters
+
+PyObject* pointToTupleBoostConverter::convert(cv::Point2f const& pt)
+{
+    PyObject* elem1 = PyFloat_FromDouble(pt.x);
+    PyObject* elem2 = PyFloat_FromDouble(pt.y);
+    PyObject* tuple = PyTuple_Pack(2, elem1, elem2);
+    return tuple;
+}
+
+pointFromTupleBoostConverter::pointFromTupleBoostConverter()
+{
+    boost::python::converter::registry::push_back(convertible, construct,
+                                                  boost::python::type_id<cv::Point2f>());
+}
+
+// check if PyObject is a tuple of numbers and can be converted to point
+void* pointFromTupleBoostConverter::convertible(PyObject* object)
+{
+    // check if we have a tuple of size 2
+    if (!PyTuple_Check(object) || PyTuple_Size(object) != 2) return NULL;
+
+    // check if the elements are numbers
+    PyObject* elem1 = PyTuple_GetItem(object, 0);
+    PyObject* elem2 = PyTuple_GetItem(object, 1);
+    if (!PyNumber_Check(elem1) || !PyNumber_Check(elem2)) return NULL;
```

Comment:
This is a bit dangerous. You check whether it is a `PyNumber`, but in the convertor, you call `PyFloat_AsDouble` on it. Not every type implementing the Python's number protocol needs to have `__float__` member function (e.g., an array).

Reply:
Well, what would you suggest? With `PyFloat_Check` a tuple like this `(10.1, 2)` could not be converted as the 2nd element is not a float (although it can be converted with `PyFlaot_AsDouble`)...

Maybe we can check for errors after calling `PyFloat_AsDouble` in `construct`, anyways all we can achieve is a bit better error message or am I wrong?